### PR TITLE
Add `--dmg-title` flag

### PR DESCRIPTION
**Pull request by dxdc**

### CODE REVIEW

**Comment by sindresorhus** [comment]
Why do you need to change this?

Per readme:

> This tool is intentionally opinionated and simple. I'm not interested in adding lots of options.

I'm happy to consider things, but there has to be a very good reason for it to exist.

**Comment by dxdc** [comment]
Hi @sindresorhus, thanks for reviewing this so quickly, and I did see that note in the readme. Great tool by the way.

> This tool is intentionally opinionated and simple. I'm not interested in adding lots of options.

My motivation for the PR is because my app name was > 27 characters, and create-dmg was failing with the error message saying it could not create the DMG because the appname (title) was too long.

I thought about some automated solutions to cure this (the simplest is to take the first 27 chars of the appname only for the DMG title), but this isn't ideal. Open to other options of course, but it seems there should be a way to handle any app name length.

File: cli.js [hunk 1]
```diff
@@ -21,6 +21,7 @@ const cli = meow(`
 	Options
 	  --overwrite         Overwrite existing DMG with the same name
 	  --identity=<value>  Manually set code signing identity (automatic by default)
+	  --dmgtitle=<value>  Manually set title of DMG volume (27 character limit)
```
Comment (sindresorhus) [inline cli.js]:
```suggestion
	  --dmg-title=<value>  Manually set title of DMG volume (27 character limit)
```

And mention that the default is the app name. And one should **only** use this if the app name is longer than 27 characters

**Comment by sindresorhus** [comment]
You're missing the most obvious fix though. Make your actual app name shorter. If you're beyond 27 characters, you should seriously consider if you have a good app name.

One part of me is against this change purely because too long app names is a bad user-experience.

**Comment by dxdc** [comment]
@sindresorhus I committed some changes to address your concerns. While I agree most apps should be shorter, there are several commercial cases where this is not the case. A lot of times they are one-off tools (like uninstallers). Here are just a few examples:

Consider:
`YubiKey Personalization Tool`
`RealLegal E-Transcript Viewer`
`Chrome Remote Desktop Host Uninstaller`
`Uninstall Adobe Photoshop CC 2019`

Truncating the appName is another option, though I think allowing the user to provide a name is superior than trying to automatically truncate it.

You can also consider the case for other users in other countries that want to use your app. German, for example, has some of the longest words in the world.

File: cli.js [hunk 2]
```diff
@@ -19,8 +19,9 @@ const cli = meow(`
 	  $ create-dmg <app> [destination]
 
 	Options
-	  --overwrite         Overwrite existing DMG with the same name
-	  --identity=<value>  Manually set code signing identity (automatic by default)
+	  --overwrite          Overwrite existing DMG with the same name
+	  --identity=<value>   Manually set code signing identity (automatic by default)
+	  --dmg-title=<value>  Manually set title of DMG volume (only used if app name is >27 character limit)
```
Comment (sindresorhus) [inline cli.js:24]: I think it should say that it defaults to the app name.

**Pull request merged**